src/storybook/functions/create-component-story.js

```diff
@@ -41,9 +44,43 @@ export function createStoryMetaSettings({ component, enumPropNamesArray, iconPro
     };
   });
 
-  actionPropsArray?.forEach(propName => {
-    argTypes[propName] = { action: propName };
+  actionPropsArray?.forEach(actionProp => {
+    if (typeof actionProp === "string") {
+      argTypes[actionProp] = { action: actionProp };
+    } else if (actionProp?.name && actionProp.linkToValueProp) {
+      // we assume that actionPropsArray is static. If it changes, things may break, since internally we call React.useState for the story decorator.
+      decorators.push(createMappedActionToInputPropDecorator(actionProp.name, actionProp.linkToValueProp));
+    }
   });
 
-  return argTypes;
+  return { argTypes, decorators };
+}
+
+/**
+ * Creates a decorator which maps a callback prop to an input prop.
+ * For example: mapping the onChange callback of a Select component to its currentValue prop.
+ * Useful for adding interactivity to stories of controlled components.
+ * Additionally, the callback will trigger a Storybook action, that can be seen on the Actions tab.
+ * @param {string} actionName - the name of the action prop of the component in the story. For example, "setValue" or "onChange".
+ * @param {string} linkToPropName - the name of the prop which should be updated when the prop of "actionName" is called. For example, "value".
+ * @returns A decorate for storybook which updates the {@link linkToPropName} input of the component, whenever {@link actionName} is called.
+ */
+function createMappedActionToInputPropDecorator(actionName, linkToPropName) {
```

Comment:
the only thing I would change is the name of the second parameter to be the same as in the createStoryMetaSettings structure. In other words, I would change "linkToPropName" to "linkToValueProp"

and maybe change the param name to: "linkedToPropValue" ?

Reply:
sure, it makes sense. I'll do it now :)